packages/flutter/lib/src/foundation/stack_frame.dart

```diff
@@ -0,0 +1,183 @@
+    if (method.startsWith('new')) {
+      className = method.split(' ')[1];
+      method = '';
+      if (className.contains('.')) {
+        final List<String> parts  = className.split('.');
+        className = parts[0];
+        method = parts[1];
+      }
+      isConstructor = true;
+    } else if (method.contains('.')) {
+      final List<String> parts = method.split('.');
+      className = parts[0];
+      method = parts[1];
+    }
+
+    final Uri packageUri = Uri.parse(match.group(3));
+    String package = '<unknown>';
+    String packagePath = packageUri.path;
+    if (packageUri.scheme == 'dart' || packageUri.scheme == 'package') {
+      package = packageUri.pathSegments[0];
+      packagePath = packageUri.path.replaceFirst(packageUri.pathSegments[0] + '/', '');
+    }
+
+    return StackFrame(
+      int.parse(match.group(1)),
+      className: className,
+      method: method,
+      packageScheme: packageUri.scheme,
+      package: package,
+      packagePath: packagePath,
+      line: int.parse(match.group(4)),
+      column: int.parse(match.group(5)),
+      isConstructor: isConstructor,
+    );
+  }
+
+  /// The zero-indexed frame number.
+  final int number;
+
+  /// The scheme of the package for this frame, e.g. "dart" for
+  /// dart:core/errors_patch.dart or "package" for
+  /// package:flutter/src/widgets/text.dart.
+  ///
+  /// The path property refers to the source file.
+  final String packageScheme;
+
+  /// The package for this frame, e.g. "core" for
+  /// dart:core/errors_patch.dart or "flutter" for
+  /// package:flutter/src/widgets/text.dart.
+  final String package;
+
+  /// The path of the file for this frame, e.g. "errors_patch.dart" for
+  /// dart:core/errors_patch.dart or "src/widgets/text.dart" for
+  /// package:flutter/src/widgets/text.dart.
+  final String packagePath;
+
+  /// The source line number.
+  final int line;
+
+  /// The source column number.
+  final int column;
+
+  /// The class name, if any, for this frame.
+  ///
+  /// This may be null for top level methods in a library or anonymous closure
+  /// methods.
+  final String className;
+
+  /// The method name for this frame.
+  ///
+  /// This will be an empty string if the stack frame is from the default
+  /// constructor.
+  final String method;
+
+  /// Whether or not this was thrown from a constructor.
+  final bool isConstructor;
+
+  @override
+  int get hashCode => hashValues(number, package, line, column, className, method);
+
+  @override
+  bool operator ==(Object other) {
+    return other is StackFrame &&
+        number == other.number &&
+        package == other.package &&
+        line == other.line &&
+        column == other.column &&
+        className == other.className &&
+        method == other.method;
+  }
+
+  @override
+  String toString() => '$runtimeType{#$number, $packageScheme:$package/$packagePath:$line:$column, className: $className, method: $method}';
```

Comment:
trivial nit: `(` instead of `{` is our convention

Reply:
Done